### PR TITLE
Add missing per-node strategy to TA CRD

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
Partially addresses #2860.

### CODE REVIEW

File: apis/v1beta1/allocation_strategy.go [hunk 1]
```diff
@@ -30,6 +30,9 @@ const (
 	// TargetAllocatorAllocationStrategyConsistentHashing targets will be consistently added to collectors, which allows a high-availability setup.
 	TargetAllocatorAllocationStrategyConsistentHashing TargetAllocatorAllocationStrategy = "consistent-hashing"
 
+	// TargetAllocatorAllocationStrategyConsistentHashingPerNode targets will be assigned to the collector on the node they reside on (use only with daemon set).
+	OpenTelemetryTargetAllocatorAllocationStrategyPerNode TargetAllocatorAllocationStrategy = "per-node"
```
Comment (pavolloffay) [inline apis/v1beta1/allocation_strategy.go:34]: It would be good to align the name with the other ones e.g. `TargetAllocatorAllocationStrategyPerNode`

**Comment by pavolloffay** [comment]
I would consider changing the conversion from code in case a new strategy is added to the v1beta1 in future to avoid failures

**Comment by pavolloffay** [comment]
it can be done in a separate PR

File: apis/v1beta1/allocation_strategy.go [hunk 2]
```diff
@@ -30,6 +30,9 @@ const (
 	// TargetAllocatorAllocationStrategyConsistentHashing targets will be consistently added to collectors, which allows a high-availability setup.
 	TargetAllocatorAllocationStrategyConsistentHashing TargetAllocatorAllocationStrategy = "consistent-hashing"
 
+	// TargetAllocatorAllocationStrategyConsistentHashingPerNode targets will be assigned to the collector on the node they reside on (use only with daemon set).
```
Comment (pavolloffay) [inline apis/v1beta1/allocation_strategy.go:33]: The comment should be fixed

**Comment by swiatekm-sumo** [comment]
> I would consider changing the conversion from code in case a new strategy is added to the v1beta1 in future to avoid failures

You mean like, writing an explicit conversion functions between the v1beta1 and v1alpha1 enums? Sure, I can do that.

**Pull request merged**